src/crypto/crypto_util.h

```diff
@@ -231,7 +231,9 @@ class ByteSource {
     }
 
     // Returns the (allocated) size in bytes.
-    size_t size() const { return size_; }
+    constexpr size_t size() const noexcept { return size_; }
+
+    constexpr bool empty() const noexcept { return size_ == 0; }
```

Comment:
What's the point in having `constexpr` and `noexcept` on these functions?

Reply:
Not anything worth measuring for now but in general I think we should use constexpr and noexcept more.

- `constexpr` is good to have, since any function that calls `empty()` can now be a `constexpr`.
- `noexcept` means it will not throw an exception. according to a google search: "The compiler doesn't need to generate the stack unwinding code, and fewer possible execution paths should be considered."

Reply:
> `constexpr` is good to have, since any function that calls `empty()` can now be a `constexpr` (and is not blocked to be constexpr).

While that's true, how does that actually help? Maybe I'm missing something, but I don't see how `*this` could exist in a `constexpr` evaluation.

> `noexcept` means it will not throw an exception. according to a google search: "The compiler doesn't need to generate the stack unwinding code, and fewer possible execution paths should be considered."

We already compile with `-fno-exceptions`, so the compiler already does not need to generate stack unwinding code in any case. Since we don't use exceptions, following this argument, we'd have to explicitly annotate every function in our repository with `noexcept`.